lib/hackerone/client/report.rb

```diff
@@ -68,17 +67,20 @@ def writeup_classification
         classification_label().split("-").first
       end
 
+      def activities
+        relationships.dig(:activities, :data)&.map do |activity_data|
+          Activity.new(activity_data)
+        end
+      end
+
       private
+
       def payments
-        activities.select { |activity| activity[:type] == PAYOUT_ACTIVITY_KEY }
+        activities.select(&:payout?)
       end
 
       def payment_amount(payment)
-        payment.fetch(:attributes, {}).fetch(:bounty_amount, "0").gsub(/[^\d]/, "").to_i
-      end
-
-      def activities
-        relationships.fetch(:activities, {}).fetch(:data, [])
+        payment.bounty_amount.gsub(/[^\d]/, "").to_i
```

Comment:
Probably not, I've chosen to create an activity class per type. It allows us to implement methods specific per type. In PR's current state I've only added the VCR recorded activities, I guess it would be nice to support all types defined used in HackerOne.